rpc/lib/server/handlers.go

```diff
@@ -532,20 +535,21 @@ func (wsc *wsConnection) writeRoutine() {
 		case <-wsc.Quit:
 			return
 		case <-wsc.pingTicker.C:
+			wsc.baseConn.SetWriteDeadline(time.Now().Add(time.Second * wsWriteTimeoutSeconds))
```

Comment:
It was a bug for it to have been missing. We defined a write timeout parameter and then forgot to apply it! I'll add a new function instead that wraps WriteMessage and includes the set deadline.